src/mjolnir/shortcutbuilder.cc

```diff
@@ -214,12 +226,16 @@ bool CanContract(GraphReader& reader,
   const DirectedEdge* oppdiredge2 = reader.GetGraphTile(oppedge2)->directededge(oppedge2);
 
   // If either opposing directed edge has exit signs return false
+  // TODO(nils): can this be removed? if there's a third outbound edge involved to exit to
+  //   we won't look at this node anyways..
   if (oppdiredge1->sign() || oppdiredge2->sign()) {
     return false;
   }
 
   // Do not allow a shortcut on a ramp crossing at a traffic signal or where
   // more than 3 edges meet.
+  // TODO(nils): why? this doesn't seem important for costing/expansion, also we only allow exactly 2
```

Comment:
this seems like the sort of thing that is added to fix some issue that was found without it. There are tons of special cases (as I remember) with ramps as they move from highway to "regular" roads.

Reply:
yeah I'm just wondering how that can have any effect on shortcut creation, since if there's a ramp, that node would have at least 3 outgoing edges and we're early-abort its  contraction

Reply:
should I work on this, I guess there's a ton of little scenarios I'll need to test